### DISCUSSION TITLE
"setting up fake worker failed cannot find module './pdf.worker.js' " error after Vercel deployment

### DISCUSSION THREAD

**Discussion by VeritusAi** [vercel/community#3159] [Help] [answered]
### Summary

I am building a web app where you can upload and chat with your pdf. I am using pdfjs-dist (module version 2.16.105) to read the pdf. Everything works fine on my localhost, but after deploying it on Vercel, I am getting this error - 

Setting up fake worker failed: "Cannot find module './pdf.worker.js'
Require stack:
- /var/task/node_modules/pdfjs-dist/legacy/build/pdf.js
- /var/task/.next/server/pages/api/process.js
- /var/task/node_modules/next/dist/server/next-server.js
- /var/task/___next_launcher.cjs".

![Screenshot 2023-07-08 104201](https://github.com/vercel/community/assets/137105523/a9f72434-4e09-41a1-903b-9c75be010976)

### Steps to Reproduce

This is the link to my web app - https://publang.vercel.app/

You can upload a pdf and then when you click on process file, you get the above mentioned error.

**Comment by amyegan** [comment, marked as answer]
Hi @Plingster. I'm not sure exactly how your project is uploading and using the PDF files, but that can be a little tricky in a serverless environment. We have a guide that might give you some clues: https://vercel.com/guides/how-can-i-use-files-in-serverless-functions

If you still need another set of eyes on this problem, please share your public repo or a minimal reproduction so we can get a better look at what is happening 🙏

**Comment by mr-chandan** [comment]
Can you share how did u fix the error it will be very helpful

**Reply by AChangXD** [reply]
@amyegan This is used to parse a pdf to text/image(pdfjs-dist is the npm package). Having the same issue atm

**Comment by zhyd1997** [comment]
import legacy pdf worker entry, can be easily found when importing the module.

**Comment by MahamadAbdulKadar** [comment]
The code worked for me is(Typescript)

`const pdfjs = await import("pdfjs-dist/legacy/build/pdf.js");`
` pdfjs.GlobalWorkerOptions.workerSrc = new URL(
      'pdfjs-dist/legacy/build/pdf.worker.js',
      import.meta.url,
    ).toString();`
